spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/MM.scala

```diff
@@ -69,36 +69,40 @@ class MM[T: ClassTag](
       output.resize(ma.size(1), mb.size(2))
       output.mm(ma, mb)
     } else {
-      require(mb.dim() == 3, "second input tensor must be 3D" +
+      require(mb.dim() == 3 || mb.dim() == 4, "second input tensor must be 3D or 4D, but get " +
         s"second input dim ${mb.dim()}")
       require(ma.size(1) == mb.size(1), "inputs must contain the same number of minibatches" +
```

Comment:
Assuming it does not work for 4D input ? For 4D input, we should compare product value of first 2 dims ?

Reply:
Dimension check is done in following code

Reply:
Does this mean that  ma.size(1) == mb.size(1) && ma.size(2) == mb.size(2) ? both need to be met ?

Reply:
ma.size(1) == mb.size(1) means batchsize should be same. but may not be same for dimension 2.  It is matrix multiply, and we will check dimension before mul.

Reply:
ok, I see. changed